File: core/ui/src/main/java/com/example/instalearnenglish/core/ui/adapter/LessonPagerAdapter.java
package com.example.instalearnenglish.core.ui.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class LessonPagerAdapter extends FragmentStateAdapter {

    public LessonPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    // 4 Tab: Tip, Vocab, Simulation, Game
    @Override
    public int getItemCount() {
        return 4;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        // This will be overridden by feature-specific adapters
        return new Fragment();
    }
}

File: core/ui/src/main/java/com/example/instalearnenglish/core/ui/adapter/LessonPagerAdapter.kt
package com.example.instalearnenglish.core.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class LessonPagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {

    // 4 Tab: Tip, Vocab, Simulation, Game
    override fun getItemCount(): Int = 4

    override fun createFragment(position: Int): Fragment {
        // Tạm thời trả về Fragment trống, Developer A/B/C sẽ thay thế bằng Fragment thật sau
        return Fragment()
    }
}
